lib/feature/one_hot_encoder.cc

```diff
@@ -73,51 +108,16 @@ llvm::Error OneHotEncoderModel::setModelData(const std::string model_data_str) {
   return llvm::Error::success();
 }
 
-namespace {
-/*
- * Given a directory path, gets the single file in the directory.
- *
- * Flink ML saves model data in a file whose name is unknown to C++. This
- * function helps to locate that file.
- *
- * This function returns empty string if the directory does not exist, or there
- * is zero or more than one file in the directory.
- */
-std::string getOnlyFileInDirectory(std::string path) {
-  std::string result = "";
-  struct dirent *entry;
-  struct stat st;
-  DIR *dir = opendir(path.c_str());
-
-  if (dir == NULL) {
-    return "";
-  }
-  while ((entry = readdir(dir)) != NULL) {
-    const string full_file_name = path + "/" + entry->d_name;
-    if (stat(full_file_name.c_str(), &st) == -1)
-      continue;
-    bool is_directory = (st.st_mode & S_IFDIR) != 0;
-    if (!is_directory) {
-      if (result.compare("")) {
-        return "";
-      }
-      result = std::string(entry->d_name);
-    }
-  }
-  closedir(dir);
-  return result;
-}
-} // anonymous namespace
-
-llvm::Expected<OneHotEncoderModel>
-OneHotEncoderModel::load(const std::string path) {
-  OneHotEncoderModel model;
+llvm::Expected<tfrt::RCReference<OneHotEncoderModel>>
+OneHotEncoderModel::load(const std::string &path, tfrt::HostContext *host) {
+  tfrt::RCReference<OneHotEncoderModel> model =
+      TakeRef(host->Construct<OneHotEncoderModel>(host));
 
-  ifstream params_input(path + "/metadata");
-  json params;
+  std::ifstream params_input(path + "/metadata");
+  nlohmann::json params;
   params << params_input;
   std::string is_droplast = params["paramMap"]["dropLast"].get<std::string>();
-  model.setDropLast(is_droplast.compare("false"));
+  model->setDropLast(is_droplast.compare("false"));
```

Comment:
`IsDropLast`'s value should be boolean. string::compare returns an integer.

Could we make the code more readable by using `is_droplast != "false"`?

Similarly, could we replace `if (!model_data_filename.compare(""))` with `if (model_data_filename == "")` to improve readability?

Implicitly type conversion in general reduce readability.